src/Endpoint/FileEndpoint.php

```diff
@@ -23,39 +22,28 @@ class FileEndpoint extends EndpointBase implements TokenAwareInterface
     use TokenAwareTrait;
     use ValidationAwareTrait;
 
-    protected string $permanentId;
-
-    public function getPermanentId(): string
+    /**
+     * @inheritDoc
+     */
+    public function __construct()
     {
-        return $this->permanentId;
+        // Disable the parent constructor. The configuration is handled by setFileUrl().
     }
 
-    public function setPermanentId(string $permanentId): self
```

Comment:
This change may be controversial, but IMO it was the cleanest way to achieve the goal.
For each endpoint, we enforce the url configuration.
However, in this case, the `endpointUrl` is passed in `setFileUrl()`, and not in `__construct()`,